drgn/helpers/linux/pid.py

```diff
@@ -123,3 +125,39 @@ def for_each_task_in_group(
     ):
         if other != task or include_self:
             yield other
+
+
+def is_kthread(t: Object) -> bool:
+    """
+    Check if a task is kernel thread.
+    """
+    if not t.mm and not task_state_to_char(t) == "Z":
+        return True
+    return False
```

Comment:
I'm torn about this implementation. It makes sense, but it's not obvious to me that it will always be correct. Even though I don't like hard-coding constants, I'm leaning towards:

```suggestion
    # This hasn't changed since Linux kernel commit 246bb0b1deb2 ("kill
    # PF_BORROWED_MM in favour of PF_KTHREAD") (in v2.6.27).
    PF_KTHREAD = 0x200000
    return bool(t.flags.value_() & PF_KTHREAD)
```

Since this seems to have come from drgn-tools, @SiddhiK29 or @brenns10 do you have any thoughts?